src/AiClient.php

```
@@ -0,0 +1,413 @@
+
+    /**
+     * Generates text using the traditional API approach.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    public static function generateTextResult($prompt, ModelInterface $model = null): GenerativeAiResult
+    {
+        return self::executeGeneration($prompt, $model, 'text');
+    }
+
+    /**
+     * Streams text generation using the traditional API approach.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return Generator<GenerativeAiResult> Generator yielding partial text generation results.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    public static function streamGenerateTextResult($prompt, ModelInterface $model = null): Generator
+    {
+        // Convert prompt to standardized Message array format
+        $messages = PromptNormalizer::normalize($prompt);
+        /** @var list<Message> $messageList */
+        $messageList = array_values($messages);
+
+        // Get model - either provided or auto-discovered
+        $resolvedModel = $model ?? ModelDiscovery::findTextModel(self::defaultRegistry());
+
+        // Validate model supports text generation
+        InterfaceValidator::validateTextGeneration($resolvedModel);
+
+        // Stream the results using the model
+        /** @phpstan-ignore-next-line */
+        yield from $resolvedModel->streamGenerateTextResult($messageList);
+    }
```

@felixarntz Since we're not starting with streaming support, should we just remove all streaming methods for now?

---

@felixarntz should I remove the streaming methods now?

---

I think they should throw an exception that they're not implemented yet, but they should be there so that it's clear we're working on it - otherwise people will ask for it. That's what we decided elsewhere too (I think it already came up in #39 or somewhere related).

---

@JasonTheAdams @felixarntz resolved here https://github.com/WordPress/php-ai-client/pull/55/commits/29097f2eeb7ee61ee2808049ee5e2887b55d29f5